gui/wxpython/datacatalog/tree.py

```diff
@@ -595,6 +597,23 @@ def OnGetItemImage(self, index, which=wx.TreeItemIcon_Normal, column=0):
         except ValueError:
             return 0
 
+    def OnGetItemNote(self, index):
+        """Overriden method to return note for each mapset.
+           Used to distinguish mapsets by lock and ownership."""
+        node = self._model.GetNodeByIndex(index)
+        node.label = node.data['name']
+        if node.data['type'] in ('mapset'):
+            mapset_path = os.path.join(node.parent.parent.data['name'],
+                                       node.parent.data['name'],
+                                       node.data['name'])
+            user = get_user_if_different(mapset_path)
+            if is_mapset_locked(mapset_path):
+                node.label = _("<{label}> (locked)").format(mapset=node.label)
+            if user:
+                node.label = _("<{label}> ({user})").format(mapset=node.label,
+                                                          user=user)
+        return node.label
+
```

Comment:
OnGetItemSomething are defined methods in wx that need to be overriden, so you can't just create a new one.

The idea here is to add more info to node.data, e.g. node.data['lock'], and then change DataCatalogNode.label method.

Reply:
No sure where to add more information to node.data. Where is e.g. node.data['name'] defined?

Reply:
> No sure where to add more information to node.data. Where is e.g. node.data['name'] defined?

I haven't delved deeply into this code, yet, but IIUC, these are defined whenever you create a dictionary for a tree node, e.g. on lines 336ff (in this PR):

```
            mapset_node = self._model.AppendNode(
                                parent=location_node,
                                data=dict(type='mapset',
                                          name=mapset,
                                          lock=is_mapset_locked(mapset_path),
                                          user=different_user))
```
So, "lock" seems to be defined for mapset nodes whenever they are created (also elsewhere in the code). But AFAICT location_nodes do not include the "lock" key.

Reply:
> OnGetItemSomething are defined methods in wx that need to be overriden, so you can't just create a new one.
> 
> The idea here is to add more info to node.data, e.g. node.data['lock'], and then change DataCatalogNode.label method.

IIUC, this is true for many other OnGetItem method that you propose.